src/operations/blas2/spmv.hpp

```diff
@@ -0,0 +1,284 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename spmv.hpp
+ *
+ **************************************************************************/
+
+#ifndef SPMV_HPP
+#define SPMV_HPP
+#include "operations/blas2_trees.h"
+#include "operations/blas_operators.hpp"
+#include "views/view_sycl.hpp"
+#include <stdexcept>
```

Comment:
Does the implementation need the `stdexcept` and `vector` includes here?

Reply:
No, I removed them.